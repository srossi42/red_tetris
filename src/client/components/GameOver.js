import React from 'react'

import Win from '../util/images/success.png'
import End from '../util/images/game-over.png'

const container = {
    position: "absolute",
    margin: 'auto',
    display: 'flex',
    justifyContent: 'center',
    height: '100%',
    alignItems: 'center'
}

const image = {
    width: '50%'
}

const GameOver = ({win}) => {
    return (
        <span style={container}>
            <img
                style={image}
                src={win ? Win : End}
                alt={win ? "gagne" : "perdu"}
                />
        </span>
    )
}

export default GameOver