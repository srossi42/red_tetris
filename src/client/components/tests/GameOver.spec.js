import React from 'react'
import { shallow } from 'enzyme';
import GameOver from '../GameOver';
import toJson from "enzyme-to-json"

describe('<GameOver/> Component', () => {
  it('rendering correctly GameOver win', () => {
    let wrapper = shallow(
      <GameOver win={true}/>
    )
    expect(toJson(wrapper)).toMatchSnapshot()
  })

  it('rendering correctly GameOver not win', () => {
    let wrapper = shallow(
      <GameOver win={false}/>
    )
    expect(toJson(wrapper)).toMatchSnapshot()
  })
})
