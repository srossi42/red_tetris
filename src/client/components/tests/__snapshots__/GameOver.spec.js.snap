// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<GameOver/> Component rendering correctly GameOver not win 1`] = `
<span
  style={
    Object {
      "alignItems": "center",
      "display": "flex",
      "height": "100%",
      "justifyContent": "center",
      "margin": "auto",
      "position": "absolute",
    }
  }
>
  <img
    alt="perdu"
    src="game-over.png"
    style={
      Object {
        "width": "50%",
      }
    }
  />
</span>
`;

exports[`<GameOver/> Component rendering correctly GameOver win 1`] = `
<span
  style={
    Object {
      "alignItems": "center",
      "display": "flex",
      "height": "100%",
      "justifyContent": "center",
      "margin": "auto",
      "position": "absolute",
    }
  }
>
  <img
    alt="gagne"
    src="success.png"
    style={
      Object {
        "width": "50%",
      }
    }
  />
</span>
`;
